### PR TITLE
Command-line arguments using argparse, whitespace .ids file fix, moved ${EXTLIBS}, small fixes.

### PR DESCRIPTION
**Pull request by moeyensj**

### CODE REVIEW

File: bin/makeLinkTrackletsInput_byNight.py [hunk 1]
```diff
@@ -76,63 +74,62 @@ def writeDetsAndIds(detsFile, idsFile, compatibleNights):
...
-            compatibleNights = sorted(filter(lambda x: x <= night
-                                             and x >= (night - WINDOW_SIZE), nights))
-        else:
-            compatibleNights = sorted(filter(lambda x: x >= night
-                                             and x <= (night + WINDOW_SIZE), nights))
         # Don't create redundant work by creating identical sets, and don't create output if
         # linkTracklets won't run (linkTracklets requires 3 nights of tracklets).
+        compatibleNights = sorted(filter(lambda x: x >= night
+                                             and x <= (night + args.windowSize), nights))
+
```
Comment (mjuric) [inline bin/makeLinkTrackletsInput_byNight.py]: Reformulate this to something like:
```
compatibleNights = sorted( x for x in nights if x >= night and x <= night + args.windowSize )
```
(but do double check that I got it right!). We're trying to move away from functional programming style (in favor of list comprehensions and iterators), as they're more readable to an average person.

**Comment by mjuric** [comment]
Hi @moeyensj; I had just one minor comment that you can fix later (see above) -- I'm merging this.

File: bin/makeLinkTrackletsInput_byNight.py (hunk 1 above)
Comment (timj) [inline bin/makeLinkTrackletsInput_byNight.py]: There are a lot of `map(lambda` lines in the codebase that could be nicely rewritten as list comprehensions. There are also many many prints that need fixing to be function (`futurize -1 -w` is your friend).

File: bin/makeLinkTrackletsInput_byNight.py [hunk 2]
```diff
@@ -76,63 +74,62 @@ def writeDetsAndIds(detsFile, idsFile, compatibleNights):
...
+    print "Reading dias files from %s" % (args.diasDir)
+    print "Reading tracklet files from %s" % (args.trackletDir)
+    print "Will write %d day window *.dets and *.ids files to %s" % (args.windowSize , args.outputDir)
 
-    print "All nights: ", nights
+    # Find the list of all detection files
+    diasFiles = glob.glob(os.path.join(args.diasDir + '*' + args.diasSuffix))
 
+    # Sort the files in MJD order.
+    nights = sorted(map(lambda x: int(x[len(args.diasDir):-(len(args.diasSuffix))]), diasFiles))
```
Comment (moeyensj) [inline bin/makeLinkTrackletsInput_byNight.py]: @mjuric @timj Considering your earlier comments, would it be appropriate for me to change this as well? To something like:

`nights = sorted(int(os.path.basename(x).split('.')[0]) for x in diasFiles)`

**Comment by moeyensj** [comment]
@mjuric I've created a non-urgent pull request for changes like the one you suggested so I can change to the preferred format when found and where needed. (I tested the code snippet you suggested and it worked without needing any changes).

File: bin/makeLinkTrackletsInput_byNight.py (hunk 2 above)
Comment (timj) [inline bin/makeLinkTrackletsInput_byNight.py]: That's certainly a lot more obvious to me in terms of sorting on the thing before the first `.`. I'd like the comment to be updated to indicate what form the filenames have that this MJD sorting is going to work.

**Pull request merged**